.github/workflows/macos-workflow.yml

```diff
@@ -115,3 +124,53 @@ jobs:
       - name: Run Tests
         working-directory: build
         run: make -j$(getconf _NPROCESSORS_ONLN) unittests
+
+      - name: Prepare Artifact Metadata
+        id: artifact-metadata
+        run: |
+          if [ ${{ github.event_name }} == "pull_request" ]; then
+            PR_SHA=$(git rev-parse --short "${{ github.event.pull_request.head.sha }}")
+            ARTIFACT_NAME="PCSX2"
+            if [ -n "${{ github.event.pull_request.number }}" ]; then
+              PR_NUM=${{ github.event.pull_request.number }}
+              ARTIFACT_NAME="${ARTIFACT_NAME}-pr[${PR_NUM}]"
+            fi
+            ARTIFACT_NAME="${ARTIFACT_NAME}-sha[${PR_SHA}]"
+            if [ -n "${{ github.event.pull_request.title }}" ]; then
+              PR_TITLE=$(echo "${{ github.event.pull_request.title }}" | tr -cd '[a-zA-Z0-9[:space:]]_-')
+              ARTIFACT_NAME="${ARTIFACT_NAME}-title[${PR_TITLE}"
+            fi
+          else
+            SHA=$(git rev-parse --short "$GITHUB_SHA")
+            ARTIFACT_NAME="PCSX2-sha[${SHA}"
+          fi
+          TRIMMED_ARTIFACT_NAME=$(printf "%.199s]" "$ARTIFACT_NAME")
+          echo "name=$TRIMMED_ARTIFACT_NAME"
+          echo "##[set-output name=name;]${TRIMMED_ARTIFACT_NAME}"
+
+      - name: Prepare Build Artifacts
+        run: |
+          cp /usr/local/lib/libMoltenVK.dylib build/pcsx2/PCSX2.app/Contents/Frameworks/
+          mv build/pcsx2/PCSX2.app "${{ steps.artifact-metadata.outputs.name }}.app"
+          tar cvzf "${{ steps.artifact-metadata.outputs.name }}.tar.gz" "${{ steps.artifact-metadata.outputs.name }}.app"
+          mkdir ci-artifacts
+          cp "${{ steps.artifact-metadata.outputs.name }}.tar.gz" ci-artifacts/macOS.tar.gz
+          ls
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.artifact-metadata.outputs.name }}
+          path: ./*.tar.gz
+
+      - name: Upload Assets and Potential Publish Release
```

Comment:
i believe this is going to make the release asset poorly named -- in the linux/windows steps there is an additional step that names them cleaner than the current PR/Commit artifact https://github.com/PCSX2/pcsx2/blob/master/.github/workflows/windows-workflow.yml#L188-L194

Additionally, should the name of the artifact include the version of macOS that is supported? I see the CI is using big sur.

Also this can be tested on a fork before merged, i can do so.

Reply:
The whole point of doing this is to make the build support all the way back to our minimum version (10.13 at the moment).  If we wanted to only support Big Sur and up, we could continue using Homebrew binaries.

I name it `macOS.tar.gz` since we only support one architecture and one SIMD ISA.  If that changes in the future, we could rename it, though ideally we'd make a fat binary and shove it all in one download.  Does that seem reasonable?

Reply:
The name of the asset will be used to make it clear to the user on the frontend what version it supports.  I'm not saying it should be labelled as big sur (although if something is ever merged that breaks on 10.13, aka high sierra?, it's probably going to go unnoticed).

If you are targetting high sierra, then we should label it as such.

Reply:
The compiler is pretty good about making sure we keep compatibility, ignoring 10.13-specific bugs, so I don't expect compatibility breaks to happen very often.

If we're going to label macOS 10.13, should we making the Windows one "Windows 8.1"?

Reply:
Yes I've thought about that argument.  From my experience with macOS apps, they usually label what version they support clearly -- I suppose this has something to do with how frequently macOS versions break previous app compatibility, but it seems to be the norm.

I guess its ultimately up to you, I have no experience building for macOS.  But the current naming this PR will commit will break the backing API's expectations as seen above -- I imagined that we would we labelling it with the version given my personal experience.

Reply:
I looked around at some other applications.  Chrome shows versions for both.  Opera does for neither.  Dolphin does for neither.  Sublime Text does for neither.  VSCode does for neither.  IntelliJ does neither.

What application were you thinking of that only shows a version for the macOS release?

Reply:
Basically almost everyone I can remember downloading for my own mac 🤷 clearly it's not as standardized as I thought.